Objects/frameobject.c

```diff
@@ -1039,16 +1063,37 @@ PyFrame_LocalsToFast(PyFrameObject *f, int clear)
                 continue;
             }
         }
+        PyObject *oldvalue = fast[i];
         if (kind & (CO_FAST_CELL | CO_FAST_FREE)) {
-            assert(PyCell_Check(fast[i]));
-            if (PyCell_GET(fast[i]) != value) {
-                if (PyCell_Set(fast[i], value) < 0) {
+            if (oldvalue != NULL) {
```

Comment:
See above.
Add why are we duplicating this code?

Reply:
What duplication do you mean?